### PR TITLE
Handle exception for aborting

### PR DESCRIPTION
**Pull request by joseivanlopez**
Needed for https://trello.com/c/jItiacnG/335-readd-system-wide-lock-in-libstorage-ng

### CODE REVIEW

File: src/ruby/yast/wfm.rb [hunk 1]
```diff
@@ -281,6 +281,15 @@ def self.call_builtin_wrapper(*args)
         e.backtrace.join("\n"))
     end
 
+    # Handles exception to abort the process
+    private_class_method def self.handle_abort_exception(e, client)
+      Builtins.y2error("Client %1 failed with '%2' (%3).\nBacktrace:\n%4",
```
Comment (jreidinger) [inline src/ruby/yast/wfm.rb]: why it is error log when it is user defined abort?

File: src/ruby/yast/wfm.rb [hunk 2]
```diff
@@ -309,6 +318,9 @@ def self.run_client(client)
       rescue SignalException => e
         handle_signal_exception(e)
         exit(16)
+      rescue AbortException => e
+        handle_abort_exception(e, client)
+        false
```
Comment (jreidinger) [inline src/ruby/yast/wfm.rb]: so it means it kill just one WFM.Call?

**Comment by ancorgs** [comment]
In general it's ok. Just make sure the rationale is clear enough in the PR description and (even more important) in the documentation of the code.

File: src/ruby/yast/wfm.rb [hunk 3]
```diff
@@ -281,6 +281,17 @@ def self.call_builtin_wrapper(*args)
         e.backtrace.join("\n"))
     end
 
+    # Handles exception to abort the process
+    #
+    # In some cases, user can request to directly abort the process (e.g., when
+    # it is not possible to acquire a lock). In that situations, the general
+    # exception handler should be avoid to not bother the user.
```
Comment (ancorgs) [inline src/ruby/yast/wfm.rb]: avoid**ed**

File: src/ruby/yast/yast.rb [hunk 4]
```diff
@@ -7,6 +7,14 @@ class FunRef; end
   class YReference; end
   class Path; end
 
+  # Custom exception class to indicate the process should abort
+  # without any more user interaction.
+  #
+  # It avoids to show the dialog to handle a general exception
+  # with options as to lauch a debugger.
```
Comment (ancorgs) [inline src/ruby/yast/yast.rb]: lauch

File: src/ruby/yast/wfm.rb [hunk 5]
```diff
@@ -309,6 +320,10 @@ def self.run_client(client)
       rescue SignalException => e
         handle_signal_exception(e)
         exit(16)
+      rescue AbortException => e
+        # Abort was requested
+        handle_abort_exception(e, client)
```
Comment (jreidinger) [inline src/ruby/yast/wfm.rb]: do you have test? I worry that this code blown up ( two params here and method want one ).

Comment (joseivanlopez) [reply src/ruby/yast/wfm.rb]: ups, I manually tested it just before to remove the param :(

I test it again

**Pull request merged**